src/AWS.Deploy.Orchestration/Utilities/EnvironmentVariableManager.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace AWS.Deploy.Orchestration.Utilities
+{
+    public interface IEnvironmentVariableManager
+    {
+        /// <summary>
+        /// Retrieves the environment variable based on the EnvironmentVariableTarget
+        /// </summary>
+        /// <param name="variable">The environment variable to look for</param>
+        /// <param name="target">Specifes where the environment variable is stored. <see cref="EnvironmentVariableTarget"/></param>
+        string? GetEnvironmentVariable(string variable, EnvironmentVariableTarget target);
+
+        /// <summary>
+        /// Stores the environment variable with the specified value. The variable is scoped to the current process
+        /// </summary>
+        void SetEnvironmentVariable(string variable, string? value);
+    }
+
+    public class EnvironmentVariableManager : IEnvironmentVariableManager
+    {
+        public string? GetEnvironmentVariable(string variable, EnvironmentVariableTarget target)
```

Comment:
I don't think we need to worry about `EnvironmentVariableTarget`. You can just call the `GetEnvironmentVariable` that doesn't take that variable which will work for both User and Machine. Otherwise when ever we call this we will have to call it for both user and machine.

Reply:
Fixed